### PR TITLE
[dashboard] ListTeams using PublicAPI

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14310
* Depends on https://github.com/gitpod-io/gitpod/pull/14301

## How to test

1. Navigate to preview, open browser console on the Network tab
2. Sing-in
3. Observe requests made to ListTeams
4. Observe dashboard works as before.

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/dashboard/src/App.tsx [hunk 1]
```diff
@@ -165,7 +169,22 @@ function App() {
...
-                const teamsPromise = getGitpodService().server.getTeams();
+                let teamsPromise: Promise<Team[]>;
+                if (usePublicApiTeamsService) {
+                    teamsPromise = teamsService.listTeams({}).then((response) => {
+                        return response.teams.map((team) => {
+                            const t: Team = {
+                                id: team.id,
+                                name: team.name,
+                                slug: team.slug,
+                                creationTime: "",
```
Comment (andrew-farries) [inline components/dashboard/src/App.tsx:180]: As we add more public APIs does it make sense to move these RPC type -> gitpod protocol type mappings to one place or are you happy to do them inline like this?

**Review by andrew-farries — APPROVED** [review]
:+1 with one non-blocking question.

File: components/dashboard/src/App.tsx (hunk 1 above)
Comment (easyCZ) [reply components/dashboard/src/App.tsx:180]: I'll eventually add a mapper which does the conversion. For this single use, I felt it's okay to just inline it and then pull out.

**Pull request merged**